rspec-tools/rspec_tools/validation/description.py

```diff
@@ -98,3 +203,18 @@ def validate_source_language(rule_language: LanguageSpecificRule):
         elif not known_highlight(pre.code['data-lang']):
           raise RuleValidationError(f'''Rule {rule_language.id} has unknown language "{pre.code['data-lang']}" in code example in section "{name}".
 Are you looking for "{highlight_name(rule_language)}"?''')
+
+def validate_optional_subsections(rule_language: LanguageSpecificRule):
+  descr = rule_language.description
+  for optional_section in list(OPTIONAL_SECTIONS.keys()):
+    resources_section = descr.find('h2', string=optional_section)
+    if resources_section is not None:
+      titles = collect_titles(resources_section, 3)
+      subsections_seen = set()
+      for title in titles:
+        name = title.text.strip()
+        if name not in OPTIONAL_SECTIONS[optional_section]:
+          raise RuleValidationError(f'Rule {rule_language.id} has a "{optional_section}" subsection with an unallowed name: "{name}"')
+        if name in subsections_seen:
+          raise RuleValidationError(f'Rule {rule_language.id} has duplicate "{optional_section}" subsections. There are 2 occurences of "{name}"')
+        subsections_seen.add(name)
```

Comment:
```suggestion
        if name not in OPTIONAL_SECTIONS[optional_section]:
          raise RuleValidationError(f'Rule {rule_language.id} has a "{optional_section}" subsection with an unallowed name: "{name}"')
        if name in subsections_seen:
          raise RuleValidationError(f'Rule {rule_language.id} has duplicate "{optional_section}" subsections. There are 2 occurences of "{name}"')
        subsections_seen.add(name)
```
This code is duplicated in `validate_how_to_fix_it_subsections` and `validate_how_to_fix_it_framework`

Reply:
Yes, it has been refactored as `validate_subsections`